source/iterator/cdc.go

```diff
@@ -0,0 +1,236 @@
+	PollingPeriod time.Duration
+	Position      *Position
+}
+
+// NewCDC creates a new instance of the [CDC].
+func NewCDC(ctx context.Context, params CDCParams) (*CDC, error) {
+	cdc := &CDC{
+		hubspotClient: params.HubSpotClient,
+		resource:      params.Resource,
+		bufferSize:    params.BufferSize,
+		pollingPeriod: params.PollingPeriod,
+		records:       make(chan sdk.Record, params.BufferSize),
+		errC:          make(chan error, 1),
+		stopC:         make(chan struct{}, 1),
+		position:      params.Position,
+	}
+
+	if cdc.position == nil || cdc.position.Timestamp == nil {
+		now := time.Now()
+		cdc.position = &Position{
+			Mode:      CDCPositionMode,
+			Timestamp: &now,
+		}
+	}
+
+	if err := cdc.loadRecords(ctx); err != nil {
+		return nil, fmt.Errorf("initial load record: %w", err)
+	}
+
+	go cdc.poll(ctx)
+
+	return cdc, nil
+}
+
+// HasNext returns a bool indicating whether the iterator has the next record to return or not.
+func (c *CDC) HasNext(ctx context.Context) (bool, error) {
+	return len(c.records) > 0, nil
+}
+
+// Next returns the next record.
+func (c *CDC) Next(ctx context.Context) (sdk.Record, error) {
+	select {
+	case <-ctx.Done():
+		return sdk.Record{}, fmt.Errorf("context cancelled: %w", ctx.Err())
+
+	case err := <-c.errC:
+		return sdk.Record{}, fmt.Errorf("async error: %w", err)
+
+	case record := <-c.records:
+		return record, nil
+	}
+}
+
+// poll polls items at the specified time intervals.
+func (c *CDC) poll(ctx context.Context) {
+	ticker := time.NewTicker(c.pollingPeriod)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+
+		case <-c.stopC:
+			return
+
+		case <-ticker.C:
+			if err := c.loadRecords(ctx); err != nil {
+				c.errC <- fmt.Errorf("load records: %w", err)
+			}
+		}
+	}
+}
+
+// loadRecords loads HubSpot items using timestamp-based filtering and search endpoints.
+// The method tries to retrieve items filtering them by updatedAfter or lastmodifieddate.
+func (c *CDC) loadRecords(ctx context.Context) error {
+	// add a millisecond here in order to skip the processed item
+	*c.position.Timestamp = c.position.Timestamp.Add(time.Millisecond)
+
+	if err := c.processUpdatedItems(ctx, *c.position.Timestamp); err != nil {
+		return fmt.Errorf("process updated items: %w", err)
+	}
+
+	return nil
+}
+
+// processUpdatedItems retrieves items that were updated after the provided timestamp.
+func (c *CDC) processUpdatedItems(ctx context.Context, updatedAfter time.Time) error {
+	if timestampResource, ok := hubspot.TimestampResources[c.resource]; ok {
+		return c.fetchTimestampBasedItems(ctx, timestampResource, updatedAfter)
+	}
+
+	if searchResource, ok := hubspot.SearchResources[c.resource]; ok {
+		return c.fetchSearchBasedItems(ctx, searchResource, updatedAfter)
+	}
+
+	return nil
+}
+
+// fetchTimestampBasedItems fetches items by filtering them by updatedAt field.
+func (c *CDC) fetchTimestampBasedItems(
+	ctx context.Context, resource hubspot.TimestampResource, updatedAfter time.Time,
```

Comment:
could you please change this style `ctx context.Context, resource hubspot.TimestampResource, updatedAfter time.Time,`
to it:
```
ctx context.Context,
resource hubspot.TimestampResource, 
 updatedAfter time.Time,
```

Reply:
in other places please do the same

Reply:
Okay, I've changed the style.

Reply:
thank you!